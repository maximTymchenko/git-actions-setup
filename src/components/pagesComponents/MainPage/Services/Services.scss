@import '../../../../styles/vars';

.services {
    &__wrapp {
        display: flex;
        justify-content: space-between;
        align-items: center;
        @include main__padding;
        background-image: url('../../../../assets/img/dots-1.svg');
        background-position: right bottom;
        background-repeat: no-repeat;
        margin-top: 50px;
        margin-bottom: 30px;

        @media screen and (max-width: $tablet) {
            margin-top: 80px;
            margin-bottom: 20px;
            padding: 0;
        }

        @media screen and (max-width: $mobile) {
            margin-top: 10px;
            margin-bottom: 10px;
            background-image: none;
        }
    }

    &__nav {
        display: flex;
        align-items: flex-start;
        justify-content: space-between;

        @media screen and (max-width: $desktop) {
            overflow-x: scroll;
            width: 100%;
            max-width: 950px;
        }

        &-btn {
            @media screen and (max-width: $tablet) {
                margin-left: 20px;
            }

            @media screen and (max-width: $mobile) {
                display: none;
            }
        }
    }
    &__company {
        margin-top: 40px;
        display: flex;
        gap: 30px;
        align-items: flex-start;

        &--wrapper {
            width: 100%;
        }

        @media screen and (max-width: $small__desktop) {
            justify-content: center;
            flex-direction: column;
        }

        @media screen and (max-width: $large__tablet) {
        }

        @media screen and (max-width: $mobile) {
            margin-left: 5px;
            margin-right: 5px;
        }
    }
}
