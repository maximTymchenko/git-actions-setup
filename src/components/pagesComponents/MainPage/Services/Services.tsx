import React, { useState, FC } from 'react'
import './Services.scss'
import Nav from '../../../common/Nav'
import Button from '../../../common/Button'
import SectionTitle from '../../../common/SectionTitle'
import spectar from '../../../../assets/img/companies/spectar.svg'
import arrowRight from '../../../../assets/img/arr-right.svg'
import walmart from '../../../../assets/img/companies/walmart.svg'
import SmallCompanyContainer from '../SmallCompanyContainer'
import LargeCompanyContainer from '../LargeCompanyContainer'
import { Link } from 'gatsby'

const Services: FC = () => {
  const menuLinks = {
    activeLink: null,
    links: [
      { id: 1, title: 'All', page: '/what-we-do/' },
      { id: 2, title: 'Web development', page: '/what-we-do/web-development/' },
      { id: 3, title: 'Cloud services', page: '/what-we-do/cloud-services/' },
      { id: 4, title: 'Team extension', page: '/what-we-do/team-extension/' },
      { id: 5, title: 'AR & VR', page: '/what-we-do/ar-&-vr/' },
      {
        id: 6,
        title: 'Retail & E-commerce',
        page: '/what-we-do/retail-and-e-commerce/',
      },
      // {
      //   id: 7,
      //   title: 'Mobile development',
      //   page: '/what-we-do/mobile-development/',
      // },
    ],
  }

  return (
    <div data-testid="services" className="services">
      <div className="services__wrapp">
        <SectionTitle type="main" title="Services and expertise" />
      </div>
      <div className="services__nav">
        <Nav type="main" links={menuLinks} linkType="link__large" />
        <div className="services__nav-btn">
          <Link to="/what-we-do/" className="link">
            <Button
              icon={arrowRight}
              type={'primary'}
              className="button"
              desc={'Discover more'}
            >
              Discover more
            </Button>
          </Link>
        </div>
      </div>
      <div className="services__company">
        <LargeCompanyContainer
          typeBG="bg-main"
          img={spectar}
          description="Building a pioneering AR product for construction"
          label="CASE STUDY"
          hashtags="Microservice architecture, Microsoft HoloLens 2, REST API, Web development, +13 more"
        />
        <div className="services__company-wrapper">
          <SmallCompanyContainer
            img={walmart}
            description="Increasing Walmart weekly ads conversion with a custom web app for Ad Tech
                    company"
            label="CASE STUDY"
            hashtags="Amazon, Retail & E-commerce, React, Web development, +13 more"
          />
        </div>
      </div>
    </div>
  )
}

export default Services
