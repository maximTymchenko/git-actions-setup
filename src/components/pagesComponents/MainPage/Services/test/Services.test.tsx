import React from 'react'
import Services from '../Services'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(<Services></Services>)
  expect(getByTestId('services')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer.create(<Services></Services>).toJSON()
  expect(tree).toMatchSnapshot()
})
