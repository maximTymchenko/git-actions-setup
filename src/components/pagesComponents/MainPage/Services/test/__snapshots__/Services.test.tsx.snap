// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="services"
  data-testid="services"
>
  <div
    className="services__wrapp"
  >
    <h2
      className="title title--main"
      data-testid="section-title"
    >
      Services and expertise
    </h2>
  </div>
  <div
    className="services__nav"
  >
    <ul
      className="nav nav--main"
    >
      <a
        className="link__margin  active__border"
        href="/what-we-do/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="nav__link"
          data-testid="nav"
          onClick={[Function]}
        >
          All
        </li>
      </a>
      <a
        className="link__margin  active__border"
        href="/what-we-do/web-development/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="nav__link"
          data-testid="nav"
          onClick={[Function]}
        >
          Web development
        </li>
      </a>
      <a
        className="link__margin  active__border"
        href="/what-we-do/cloud-services/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="nav__link"
          data-testid="nav"
          onClick={[Function]}
        >
          Cloud services
        </li>
      </a>
      <a
        className="link__margin  active__border"
        href="/what-we-do/team-extension/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="nav__link"
          data-testid="nav"
          onClick={[Function]}
        >
          Team extension
        </li>
      </a>
      <a
        className="link__margin  active__border"
        href="/what-we-do/ar-&-vr/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="nav__link"
          data-testid="nav"
          onClick={[Function]}
        >
          AR & VR
        </li>
      </a>
      <a
        className="link__margin  active__border"
        href="/what-we-do/retail-and-e-commerce/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="nav__link"
          data-testid="nav"
          onClick={[Function]}
        >
          Retail & E-commerce
        </li>
      </a>
      <a
        className="link__margin  active__border"
        href="/what-we-do/mobile-development/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <li
          className="nav__link"
          data-testid="nav"
          onClick={[Function]}
        >
          Mobile development
        </li>
      </a>
    </ul>
    <div
      className="services__nav-btn"
    >
      <a
        className="link"
        href="/what-we-do/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <button
          className="button button--primary button--undefined button--undefined"
          data-testid="button"
        >
          Discover more
          <img
            alt="Discover more"
            className="button__icon"
            data-testid="button__icon"
            src="test-file-stub"
          />
        </button>
      </a>
    </div>
  </div>
  <div
    className="services__company"
  >
    <div
      className="large-company-container__wrapper"
      data-testid="large-company-container"
    >
      <div
        className="large-company-container large-company-container--bg-main large-company-container--undefined"
      >
        <img
          alt="Spectar"
          className="large-company-container__logo"
          src="test-file-stub"
        />
        <p
          className="large-company-container__desc"
        >
          Building a pioneering AR product for construction
        </p>
      </div>
      <p
        className="large-company-container__hashtags large-company-container__hashtags--undefined"
      >
        &gt;
        <span
          className="large-company-container__label"
        >
          CASE STUDY
        </span>
        Microservice architecture, Microsoft HoloLens 2, REST API, Web development, +13 more
      </p>
    </div>
    <div
      className="small-company-container__wrapper"
      data-testid="small-company-container"
    >
      <div
        className="small-company-container small-company-container--undefined"
      >
        <img
          alt="Walmart"
          className="small-company-container__logo"
          src="test-file-stub"
        />
        <p
          className="small-company-container__desc"
        >
          Increasing Walmart weekly ads conversion with a custom web app for Ad Tech company
        </p>
      </div>
      <p
        className="small-company-container__hashtags small-company-container__hashtags--undefined"
      >
        <span
          className="small-company-container__label"
        >
          CASE STUDY
        </span>
        Amazon, Retail & E-commerce, React, Web development, +13 more
      </p>
    </div>
  </div>
</div>
`;
